src/lightning_trainable/datasets/toy/utils.py

```diff
@@ -14,3 +14,13 @@ def sample_sphere(sample_shape: tuple[int], dimensions: int, radii: torch.Tensor
     points = radii * points / torch.linalg.norm(points, dim=-1, keepdim=True)
 
     return points.squeeze(-1)
+
+
+def sample_cube(samples: int, dimensions: int, side_length: torch.Tensor):
+    """ Sample points on the corners of a hypercube """
+    assert side_length.dim() == 1
+    assert samples <= 2 ** dimensions, "Too many samples for a hypercube of this dimensionality"
```

Comment:
Should use a `ValueError` here (I did this incorrectly in other parts of the library).

Reply:
Sure! I was just following your style. Can do it tomorrow or you can do it, if you get to it earlier.